spec/ddtrace/utils_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'concurrent-ruby'
+
+RSpec.describe Datadog::Utils do
+  describe '.next_id' do
+    subject(:next_id) { described_class.next_id }
+
+    it 'returns a positive integer smaller than 2**62' do
+      is_expected.to be_a(Integer)
+      is_expected.to be_between(1, 2**62 - 1)
+    end
+
+    it 'fits in a CRuby VALUE slot', if: ObjectSpaceHelper.estimate_bytesize_supported? do
+      expect(ObjectSpaceHelper.estimate_bytesize(next_id)).to eq(0)
+    end
+
+    it 'returns unique numbers on successive calls' do
+      is_expected.to_not eq(described_class.next_id)
+    end
+
+    it 'returns unique numbers in a threaded environment' do
+      barrier = Concurrent::CyclicBarrier.new(100)
+      threads = Array.new(100) do
+        Thread.new do
+          barrier.wait
+          described_class.next_id
+        end
+      end
+
+      ids = threads.map(&:value).uniq
+      expect(ids).to have(100).items
+    end
+
+    context 'after forking', if: PlatformHelpers.supports_fork? do
+      it 'generates unique ids across forks' do
+        ids = Array.new(3) do
+          result = expect_in_fork { puts next_id }
+          result[:stdout].to_i
```

Comment:
> Will do, I'll to_i it, it should suffice today.

```suggestion
          Integer(result.fetch(:stdout))
```

because:

```
[3] pry(main)> nil.to_i
=> 0
[4] pry(main)> "".to_i
=> 0
[5] pry(main)> "     ".to_i
=> 0
[6] pry(main)> "😭".to_i
=> 0

[10] pry(main)> Integer(nil)
TypeError: can't convert nil into Integer
from (pry):10:in `Integer'
[11] pry(main)> Integer("")
ArgumentError: invalid value for Integer(): ""
from (pry):11:in `Integer'
[12] pry(main)> Integer("😅")
ArgumentError: invalid value for Integer(): "😅"
from (pry):12:in `Integer'

Reply:
I considered this, both will fail in this tests. ([0,0,0].uniq won't have 3 items and fail the assertion).

I'll change it to Integer to have an earlier error.

Reply:
> I considered this, both will fail in this tests. ([0,0,0].uniq won't have 3 items and fail the assertion).
>
> I'll change it to Integer to have an earlier error.

You're right indeed 😅 although yeah I'd still favor the more explicit thing, thanks for updating it.